packages/chat-component-bindings/src/errorBarSelector.ts

```diff
@@ -20,8 +21,64 @@ export const errorBarSelector = createSelector([getLatestErrors], (latestErrors)
   //
   // We chose to stable sort by error type: We intend to show only a small number of errors on the UI and we do not
   // have timestamps for errors.
-  if (latestErrors['ChatThreadClient.sendMessage'] !== undefined) {
-    return { activeErrors: ['sendMessageGeneric'] };
+  const activeErrors: ErrorType[] = [];
+  let specificSendMessageErrorSeen = false;
+  if (hasUnableToReachBackendError(latestErrors)) {
+    activeErrors.push('unableToReachChatService');
   }
-  return { activeErrors: [] };
+  if (hasAccessDeniedError(latestErrors)) {
+    activeErrors.push('accessDenied');
+  }
+  if (hasNotInThisThreadError(latestErrors)) {
+    if (latestErrors['ChatThreadClient.sendMessage'] !== undefined) {
+      activeErrors.push('sendMessageNotInThisThread');
+      specificSendMessageErrorSeen = true;
+    } else {
+      activeErrors.push('notInThisThread');
+    }
+  }
+  if (!specificSendMessageErrorSeen && latestErrors['ChatThreadClient.sendMessage'] !== undefined) {
+    activeErrors.push('sendMessageGeneric');
+  }
+  return { activeErrors: activeErrors };
 });
+
+const accessErrorTargets: ChatErrorTargets[] = [
```

Comment:
I remember we generate this Targets type somewhere. Is still there and is it possible to import it?

Reply:
The type is imported. But we have no way of automatically generating the values in the enum at runtime (because of uglification etc.)

Also, this list is a small subset of all the values in the type -- only the ones we usually see when there's an access problem.